### ISSUE TITLE
Use the CLI styleext to generate the correct css format

### ISSUE THREAD

**Issue by tobi-or-not-tobi** [angular/material2#10928]
#### Bug, feature request, or proposal:
feature

#### What is the expected behavior?
When we've configured the CLI to use another css format (i.e. scss) for generated components, we would expect the material schematics to align with this when generating components (such as the materialNav).

#### What is the current behavior?
The generator generates plain css whenever the styleext is set to scss. 

#### What are the steps to reproduce?
ng g @angular/material:materialNav --name=navigation/main-nav

#### What is the use-case or motivation for changing an existing behavior?
Alignment with CLI, meet developer expectations.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Angular CLI: 6.0.0-rc.5
@angular/cdk 6.0.0-rc.12
@angular/material 6.0.0-rc.12

**Comment by jelbourn** [comment]
@DevVersion want to take a look at this?

**Comment by DevVersion** [comment]
The default style extension seems to be tied to the `@angular:component` schematic.

Meaning that we aren't able to access the default options for that schematic because our Material schematic's are part of a different collection and have a different schematic name.

_[see for example here](https://github.com/angular/angular-cli/blob/a0978b1db3642bda553ce70e89ae201217d5dfd7/packages/angular/cli/commands/generate.ts#L41-L44)_
```ts
const schematicOptions = await this.getOptions({
  schematicName,
  collectionName,
});
```

@hansl Is there any way we can get access from a custom schematic, to that default schematic configuration in order to figure out the `styleext`. It feels a bit wrong to set the `styleext` for a single schematic, while the user specified it for the whole app using `ng new XX --style scss`

**Comment by amcdnl** [comment]
@DevVersion - Not sure if this helps but our component schematics were refactored into a base util [here](https://github.com/angular/material2/blob/3b50e44bf2c44b27d8dac90f42bc004648c29de3/src/lib/schematics/utils/devkit-utils/component.ts#L116-L182) and that was copied from the base component schematic [here](https://github.com/angular/angular-cli/tree/master/packages/schematics/angular/component).

**Comment by DevVersion** [comment]
@amcdnl Yeah, unfortunately it doesn't look like we could modify the logic to access the config though. The reason is that the default `styleext` is tied to **one** specific schematic with a unique namespace & name.

```ts
"schematics": {
  "@schematics/angular:component": {
    "styleext": "scss"
  }
},
```

**Comment by DevVersion** [comment]
@hansl Okay that's reasonable. Just good to know that there is nothing we could do yet. Thanks for the response.
